tasks/LocationCategoriesTask.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace Dynamic\Locator\Tasks;
+
+use Dynamic\Locator\Location;
+use SilverStripe\Dev\BuildTask;
+use SilverStripe\ORM\DataObject;
+
+/**
+ * Class LocationCategoriesTask
+ * @package Dynamic\Locator\Tasks
+ */
+class LocationCategoriesTask extends BuildTask
+{
+
+    /**
+     * @var string
+     */
+    protected $title = 'Location categories to many_many';
+
+    /**
+     * @var string
+     */
+    protected $description = 'Migration task - Converts locations to have multiple categories';
+
+    /**
+     * @var bool
+     */
+    protected $enabled = true;
+
+    /**
+     * @param $request
+     */
+    public function run($request)
+    {
+        /** @var DataObject $class */
+        $class = ($request->getVar('locationclass')) ? $request->getVar('locationclass') : Location::class;
+        $class::add_extension(LocationCategoryExtension::class);
+
+        $ct = 0;
+
+        $convert = function (DataObject $location) use (&$ct) {
```

Comment:
Let's add some checks for instances where a location might not have a category. Could skip the add code then. Also work in some memoization to reduce the number of calls to the db for the categories.